arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -104,6 +90,24 @@ class PerformValveIsolationTraceViewController: UIViewController {
         return parameters
     }
     
+    /// Load the service geodatabase and initialize the layers.
+    func loadServiceGeodatabase() {
+        let serviceGeodatabase = AGSServiceGeodatabase(url: featureServiceURL)
```

Comment:
Methinks this would need to be a property rather than an instance variable as the way it is now, only the completion handler is keeping the geodatabase around, which isn't great. The completion handler should access the geodatabase either through a property of `self` or as an unowned reference.

Reply:
This change required me to adjust the way the credentials are handled. I also added an additional status message because there were times when the viewpoint took a while to adjust and the status bar said "utility network loaded" which was confusing.

Reply:
I'm a bit confused here - why the credentials handling need to be changed? Could we still apply the credentials to the UN or even to the geodatabase? `serviceGeodatabase.credential = ...`

Reply:
By making the `serviceGeodatabase` a property, authentication is prompted twice – once for `serviceGeodatabase` and then again for `utilityNetwork`. Using the `AGSAuthenticationManager` allows me to set the credentials once and it will apply the credentials whenever prompted. Otherwise, I'd have to set the same credentials twice for separate instance variables.

Reply:
I tried to set the credential on geodatabase and not the UN, and it doesn't prompt me for credentials. This is why I asked...